### ISSUE TITLE
c7n-org: Variable(s) in JMESPath/JSON string

### ISSUE THREAD

**Issue by sjoerd-dijkstra** [cloud-custodian/cloud-custodian#6175]
I'm trying to get multiple variables within a json string to interpolate, without succes.

My attempt:

accounts.yaml
```yaml
vars:
  var_1: www.foo.bar
  var_2: No succes
```

policy.yaml
```yaml
actions:
- type: webhook
  url: "{var_1}"                 #  << works fine :)
  batch: true
  body: "{ title: '{var_2}' }".  #  << does not work :(
```

The first variable expands fine. Unfortunately the second variables does not. My guess would be that the outer curly braces of the json string (JMESPath) somehow interfere with the placeholders within the string.

How to fix or work around this issue?

**Comment by sjoerd-dijkstra** [comment]
## TL;DR

Solution by double quoting `{{` and `}}`

```yaml
actions:
- type: webhook
  url: "{var_1}"                  #  << works fine :)
  batch: true
  body: "{{ title: '{var_2}' }}"  #  << this works with double outer braces! ;)
```

## Background:

After some investigation; the problem can be solved with double quoting the outer braces.

Variable interpolation in c7n is done via the native python [String format()](https://www.w3schools.com/python/ref_string_format.asp) method. As found here:

https://github.com/cloud-custodian/cloud-custodian/blob/fcf1d22d8a4880102c0b0d4f9107a08962ebee44/c7n/policy.py#L1110-L1116

and here:

https://github.com/cloud-custodian/cloud-custodian/blob/fcf1d22d8a4880102c0b0d4f9107a08962ebee44/c7n/utils.py#L484-L505

Now when dealing with json formatted strings you're already wrapping everything in braces. Thus making it impossible for the interpreter to interpolate the variables within that section. There is a fix for this however. 

Here's the [relevant part](https://docs.python.org/3/library/string.html#formatstrings) of the Python documentation for format string syntax:

> Format strings contain “replacement fields” surrounded by curly braces {}. Anything that is not contained in braces is considered literal text, which is copied unchanged to the output. If you need to include a brace character in the literal text, it can be escaped by doubling: {{ and }}.

Stack overflow [reference](https://stackoverflow.com/questions/5466451/how-can-i-print-literal-curly-brace-characters-in-python-string-and-also-use-fo).

With double braces interpolates works fine. Even in more complex json/jmespath objects, e.g.:

```yaml
    type: webhook
    url: "{WebhookSlack}"
    batch: true
    body: "{{
        text:'Scheduled infrastructure scan found violations...'
        attachments:[
          {{
            author_name:policy.name,
            title:policy.description,
            color:'danger',
            author_icon:'https://upload.wikimedia.org/wikipedia/commons/thumb/8/8f/Orange_lambda.svg/980px-Orange_lambda.svg.png',
            fields:[
              {{
                title:'Account',
                value:join('',['`', account_id, '`']),
                short:'true'
              }},
              {{
                title:'Environment',
                value:join('',['`', {Environment}, '`']),
                short:'true'
              }},
              {{
                title:'Resources',
                value:resources[*].[{ResourceJMESPathList}] | [] | join('',['`',join('` ,  `',[]),'`'])
                short:'false'
              }}
            ]
          }}
        ]
      }}"
```

Closing the issue as i found a solution!
